### ISSUE TITLE
Fix `selector-pseudo-class-no-unknown` false positives for all resource state pseudo-classes

### ISSUE THREAD

**Issue by Mouvedia** [stylelint/stylelint#7477]
### What minimal example or steps are needed to reproduce the bug?

```css
:seeking {}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "selector-pseudo-class-no-unknown": true
  }
}
```

### How did you run Stylelint?

[https://stylelint.io/demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcCAOAbAhgTwJYDsDmABMAL4A6O8AzjDANa6EkgA0IAZlijAHJoC2cRDAAe-VDAB0YSpRbgIODngQhg5AgVIgATgFculLfCLqNmkNS5gALhG0BaJNV1QI9sOhn2cb3Tlo+AO44RgTWejCmZDjEcpCKWHgAYnZ8aNYqAFaUCnKwTipqOBpalNYYXCi41qFa6NYwZVrMpqXlldXuCkr2ZWg4UGjaULUg9Y01IOSxxEA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
  "stylelint-config-standard": "latest"
}
```

### Details

https://drafts.csswg.org/selectors/#resource-pseudos
2 out of 7 are currently supported:

- [x] playing
- [x] paused
- [x] seeking
- [x] stalled
- [x] buffering
- [x] volume-locked
- [x] muted

It's been supported by Safari since version 15.4.
see
https://caniuse.com/mdn-css_selectors_seeking
https://caniuse.com/mdn-css_selectors_stalled
https://caniuse.com/mdn-css_selectors_buffering
https://caniuse.com/mdn-css_selectors_volume-locked
https://caniuse.com/mdn-css_selectors_muted

**Comment by ybiquitous** [comment]
@Mouvedia Thanks for the report. I didn't know that properties. 👍🏼

Also, I've updated the demo link to confirm the problems quickly.

Supporting them looks good to me, but is there any difference between `:playing` and `:seeking`? They both looks the same status on MDN:

- https://developer.mozilla.org/en-US/docs/Web/CSS/:playing
- https://developer.mozilla.org/en-US/docs/Web/CSS/:seeking

**Comment by Mouvedia** [comment]
> …but is there any difference between `:playing` and `:seeking`?

Seeking is when you move the "thumb" of the "range" of an audio or video tag.
i.e. click + hold + slide

see
https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-slider-thumb
https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/seeking_event

**Comment by ybiquitous** [comment]
Got it. I misunderstood that `:seeking` is still experimental.

Adding the unsupported pseudo-classes is not a problem for me. 👌🏼

**Comment by ybiquitous** [comment]
Let's add the rest of the classes to the `pseudoClasses` reference:

https://github.com/stylelint/stylelint/blob/34cdc97149234063c94301cd64ceb378b20a07bf/lib/reference/selectors.mjs#L254

It may be also good to extract a new group for resource state pseudo-classes like this, though it's not required:

```js
// https://drafts.csswg.org/selectors/#resource-pseudos
const resourceStatePseudoClasses = new Set([
  'playing',
  // ...
]);
```
